### DISCUSSION TITLE
Create GitHub Page via REST API v3

### DISCUSSION THREAD

**Discussion by philipsberufsausbildung** [community/community#24469] [Apps, API and Webhooks]
<p>Hello everyone,</p>
<p>is it possible to create a <em>new</em> Github Page through the REST API v3?</p>
<p>I tried to use “Request a page build”, but it only works on already existing pages. If I use this endpoint for repositories where no page exists yet, I get this:</p>
<pre><code>Status: 403 Forbidden

{
	"message": "The repository does not have a GitHub Pages site. See https://developer.github.com/v3/repos/pages/",
	"documentation_url": "https://developer.github.com/v3/repos/pages/#request-a-page-build"
}
</code></pre>
<p>Do you have some ideas?</p>
<p>Thanks in advance!</p>

**Comment by lee-dohm** [comment]
<p>If you want to enable Pages for a given repository, you have to tell it which branch (and optionally if you want to use the <code>/docs</code> directory) to build the Pages site from. (Just like <a href="https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/" rel="nofollow noopener">you would have to do through the web UI</a>.) You can do this with the <a href="https://developer.github.com/v3/repos/pages/#update-information-about-a-pages-site" rel="nofollow noopener">Update information about a Pages site</a> endpoint.</p>
<p>I hope that helps!</p>

**Comment by philipsberufsausbildung** [comment]
<p>Hey <a class="mention" href="/lee-dohm">@lee-dohm</a>,</p>
<p>thanks for your reply, it sounds very useful!</p>
<p>I will try this and keep you updated!</p>

**Comment by philipsberufsausbildung** [comment]
<p>I tried your solution, but this endpoint also seems to work only with already existing pages.</p>
<p>If I create a new repository and upload a new index.html, I get a 404 Not Found at this endpoint.</p>
<p>And witch witch value should I send for “cname”? Is it [username].github.io ?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If the edit endpoint doesn’t offer what you want, then you’ll have to use the UI to initially enable Pages support for the repo. You probably shouldn’t be changing the CNAME value via the API but rather follow the system <a href="https://help.github.com/articles/setting-up-a-custom-subdomain/" rel="nofollow noopener">described in the help</a>.</p>
